### PR TITLE
JSON test data: use team name as giver in team responses #7283

### LINKED ISSUE #7283: JSON test data: use team name as giver in team responses
For team responses, the response giver should be the team name rather than the email of the student that gave the response. E.g. `response1ForQ1S2C1` in `typicalDataBundle.json`. I've only checked `typicalDataBundle.json` but it's probably the case in other `json` files too.

### PR DESCRIPTION
**Pull request by amarlearning**
Fixes #7283  

One more test remaining `FeedbackSessionResultsTest.json` will update the PR asap.

### CODE REVIEW

**Comment by amarlearning** [comment]
ready for partial review

File: src/test/java/teammates/test/cases/logic/FeedbackSessionsLogicTest.java [hunk 1]
```diff
@@ -393,8 +393,8 @@ private void testGetFeedbackSessionDetailsForInstructor() throws Exception {
 
         // 2 instructors, 6 students = 8
         assertEquals(8, stats.expectedTotal);
-        // 1 instructor, 1 student, did not respond => 8-2=6
-        assertEquals(6, stats.submittedTotal);
+        // 1 instructor, 5 student and 2 Team responded => 1+5+2=8
```
Comment (amarlearning) [inline src/test/java/teammates/test/cases/logic/FeedbackSessionsLogicTest.java]: done :+1:

**Comment by amarlearning** [comment]
> Looks okay so far, but as mentioned there's a need to change `FeedbackSessionDetailsTest.json`

@chowyb According to me all required changes had been made to `FeedbackSessionDetailsTest.json`, did you find anything missing?

**Comment by chowyb** [comment]
From `FeedbackSessionDetailsTest.json`:

```json
    "qn7.resp1": {
      "feedbackSessionName": "Standard feedback session",
      "courseId": "FSDTest.idOfTypicalCourse1",
      "feedbackQuestionId": "7",
      "feedbackQuestionType": "TEXT",
      "giver": "FSDTest.student1InCourse1@gmail.tmt",
      "recipient": "FSDTest.student2InCourse1@gmail.tmt",
      "responseMetaData": {
        "value": "team 1.1 > student (in own team)"
      }
    },
    "qn7.resp2": {
      "feedbackSessionName": "Standard feedback session",
      "courseId": "FSDTest.idOfTypicalCourse1",
      "feedbackQuestionId": "7",
      "feedbackQuestionType": "TEXT",
      "giver": "FSDTest.student2InCourse1@gmail.tmt",
      "recipient": "FSDTest.student4InCourse1@gmail.tmt",
      "responseMetaData": {
        "value": "team 1.1 > student (in Team 1.2)"
      }
```

From `FeedbackSessionResultsTest.json`:

```json
    "qn7.resp1": {
      "feedbackSessionName": "Standard feedback session",
      "courseId": "FSRTest.idOfTypicalCourse1",
      "feedbackQuestionId": "7",
      "feedbackQuestionType": "TEXT",
      "giver": "FSRTest.student1InCourse1@gmail.tmt",
      "recipient": "FSRTest.student2InCourse1@gmail.tmt",
      "responseMetaData": {
        "value": "team 1.1 > student (in own team)"
      },
      "giverSection": "Section A",
      "recipientSection": "Section A"
    },
    "qn7.resp2": {
      "feedbackSessionName": "Standard feedback session",
      "courseId": "FSRTest.idOfTypicalCourse1",
      "feedbackQuestionId": "7",
      "feedbackQuestionType": "TEXT",
      "giver": "FSRTest.student2InCourse1@gmail.tmt",
      "recipient": "FSRTest.student4InCourse1@gmail.tmt",
      "responseMetaData": {
        "value": "team 1.1 > student (in Team 1.2)"
      },
      "giverSection": "Section A",
      "recipientSection": "Section B"
    },
    "qn8.resp1": {
      "feedbackSessionName": "Standard feedback session",
      "courseId": "FSRTest.idOfTypicalCourse1",
      "feedbackQuestionId": "8",
      "feedbackQuestionType": "TEXT",
      "giver": "FSRTest.student1InCourse1@gmail.tmt",
      "recipient": "Team 1.2",
      "responseMetaData": {
        "value": "team 1.1 > team 1.2"
      },
      "giverSection": "Section A",
      "recipientSection": "Section B"
    },
    "qn8.resp2": {
      "feedbackSessionName": "Standard feedback session",
      "courseId": "FSRTest.idOfTypicalCourse1",
      "feedbackQuestionId": "8",
      "feedbackQuestionType": "TEXT",
      "giver": "FSRTest.student4InCourse1@gmail.tmt",
      "recipient": "Team 1.1</td></div>'\"",
      "responseMetaData": {
        "value": "team 1.2 > team 1.1"
      },
      "giverSection": "Section B",
      "recipientSection": "Section A"
    }
```

**Comment by amarlearning** [comment]
> For team responses, the response giver should be the team name rather than the email of the student

@chowyb this is what the issue says :arrow_up: 

As you mentioned [here](https://github.com/TEAMMATES/teammates/pull/7957#issuecomment-324238674) in `FeedbackSessionDetailsTest.json`

```json
    "qn7.resp1": {
      "feedbackSessionName": "Standard feedback session",
      "courseId": "FSDTest.idOfTypicalCourse1",
      "feedbackQuestionId": "7",
      "feedbackQuestionType": "TEXT",
      "giver": "FSDTest.student1InCourse1@gmail.tmt",
      "recipient": "FSDTest.student2InCourse1@gmail.tmt",
      "responseMetaData": {
        "value": "team 1.1 > student (in own team)"
      }
    },
    "qn7.resp2": {
      "feedbackSessionName": "Standard feedback session",
      "courseId": "FSDTest.idOfTypicalCourse1",
      "feedbackQuestionId": "7",
      "feedbackQuestionType": "TEXT",
      "giver": "FSDTest.student2InCourse1@gmail.tmt",
      "recipient": "FSDTest.student4InCourse1@gmail.tmt",
      "responseMetaData": {
        "value": "team 1.1 > student (in Team 1.2)"
      }
```
In both the above responses the `recipient` is a `student`, so giver can also be a `student`. Please correct me if I am wrong anywhere.

**Comment by chowyb** [comment]
@amarlearning I believe the issue is intended to target responses where the giver takes the role of a team, not necessarily where both the giver and the receiver take the role of a team.

It may still be possible that your interpretation of the issue description holds however (perhaps if the issue intends to fix all instances where the giver is a team, we can fix the remaining ones in a future PR). If so, then you will still need to change the following in `FeedbackSessionResultsTest.json`, which involves a team -> team question:

```json
    "qn8.resp1": {
      "feedbackSessionName": "Standard feedback session",
      "courseId": "FSRTest.idOfTypicalCourse1",
      "feedbackQuestionId": "8",
      "feedbackQuestionType": "TEXT",
      "giver": "FSRTest.student1InCourse1@gmail.tmt",
      "recipient": "Team 1.2",
      "responseMetaData": {
        "value": "team 1.1 > team 1.2"
      },
      "giverSection": "Section A",
      "recipientSection": "Section B"
    },
    "qn8.resp2": {
      "feedbackSessionName": "Standard feedback session",
      "courseId": "FSRTest.idOfTypicalCourse1",
      "feedbackQuestionId": "8",
      "feedbackQuestionType": "TEXT",
      "giver": "FSRTest.student4InCourse1@gmail.tmt",
      "recipient": "Team 1.1</td></div>'\"",
      "responseMetaData": {
        "value": "team 1.2 > team 1.1"
      },
      "giverSection": "Section B",
      "recipientSection": "Section A"
    }
```

**Comment by wkurniawan07** [comment]
Closing due to an inactivity for a very long period. If anyone is interested to continue the work done here, feel free to reopen/request for reopen.

**Pull request closed without merging**